dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/config/ServiceConfig.java

```diff
@@ -249,52 +253,68 @@ public RetryTemplate retryTemplate()
-                getCheckerByClass( ProgramStageInstanceAuthCheck.class ),
-                getCheckerByClass( AttributeOptionComboCheck.class ),
-                getCheckerByClass( AttributeOptionComboDateCheck.class ),
-                getCheckerByClass( EventGeometryCheck.class ),
-                getCheckerByClass( DataValueCheck.class ),
-                getCheckerByClass( FilteredDataValueCheck.class ),
-                getCheckerByClass( ExpirationDaysCheck.class ) ),
-            DELETE, newArrayList(
-                getCheckerByClass(
-                    DeleteProgramStageInstanceAclCheck.class ) )
+        return ImmutableList.of(
+            getInsertChecker( EventDateCheck.class ),
+            getInsertChecker( OrgUnitCheck.class ),
+            getInsertChecker( SharedProgramCheck.class ),
+            getInsertChecker( ProgramStageCheck.class ),
+            getInsertChecker( TrackedEntityInstanceCheck.class ),
+            getInsertChecker( ProgramInstanceCheck.class ),
+            getInsertChecker( ProgramInstanceRepeatableStageCheck.class ),
+            getInsertChecker( ProgramOrgUnitCheck.class ),
+            getInsertChecker( EventGeometryCheck.class ),
+            getInsertChecker( EventCreationAclCheck.class ),
+            getInsertChecker( EventBaseCheck.class ),
+            getInsertChecker( AttributeOptionComboCheck.class ),
+            getInsertChecker( AttributeOptionComboDateCheck.class ),
+            getInsertChecker( AttributeOptionComboAclCheck.class ),
+            getInsertChecker( DataValueCheck.class ),
+            getInsertChecker( FilteredDataValueCheck.class ),
+            getInsertChecker( DataValueAclCheck.class ),
+            getInsertChecker( ExpirationDaysCheck.class ) );
+    }
+
+    @Bean
+    public List<UpdateChecker> checkersRunOnUpdate()
+    {
+        return ImmutableList.of(
+            getUpdateChecker( EventSimpleCheck.class ),
+            getUpdateChecker( EventBaseCheck.class ),
+            getUpdateChecker( ProgramStageInstanceBasicCheck.class ),
+            getUpdateChecker( UpdateProgramStageInstanceAclCheck.class ),
+            getUpdateChecker( SharedProgramCheck.class ),
+            getUpdateChecker( ProgramInstanceCheck.class ),
+            getUpdateChecker( ProgramStageInstanceAuthCheck.class ),
+            getUpdateChecker( AttributeOptionComboCheck.class ),
+            getUpdateChecker( AttributeOptionComboDateCheck.class ),
+            getUpdateChecker( EventGeometryCheck.class ),
+            getUpdateChecker( DataValueCheck.class ),
+            getUpdateChecker( FilteredDataValueCheck.class ),
+            getUpdateChecker( ExpirationDaysCheck.class ) );
+    }
 
-        );
+    @Bean
+    public List<DeleteChecker> checkersRunOnDelete()
+    {
+        return ImmutableList.of(
+            getDeleteChecker( DeleteProgramStageInstanceAclCheck.class ) );
+    }
+
+    private InsertChecker getInsertChecker( Class<? extends InsertChecker> k )
```

Comment:
not that it matters here but generally `k` is a bit short for a parameter - `key` maybe